File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    answer = ''
    
    str_numbers = [str(num) for num in numbers]
    str_numbers.sort(key=lambda x : x * 3, reverse = True)
    result = ''.join(str_numbers)
    
    return result if int(result) != 0 else '0'

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        /*
        
        */
        int n = numbers.length;
        String[] arr = new String[n];
        
        for (int i = 0; i < n; i++) {
            arr[i] = String.valueOf(numbers[i]);
        }
        
        Arrays.sort(arr, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));
        
        if (arr[0].equals("0")) return "0";
        
        return String.join("", arr);
    }
}